lib/CL/devices/devices.c

```diff
@@ -689,10 +682,6 @@ pocl_init_devices ()
 #endif
 
 #ifdef ENABLE_SIGNAL_HANDLERS
-  if (pocl_get_bool_option ("POCL_SIGFPE_HANDLER", 1))
-    {
-      pocl_install_sigfpe_handler ();
```

Comment:
Do we really want to remove the old solution? As far as I see, there's quite a bit of injected code which affects the performance. Perhaps for Linux the signal handler-trapping one is faster and enables better vectorization?

Reply:
I doubt div/mod-by-zero etc. occurs in performance critical cases, it's like a bug ~always, the exception should not happen thus no overhead in the common case. If the vectorization performance is not improved in x86, then the scalar performance should be as you add those comparisons there. How much in real world benchmarks, likely not much, but if it's easy to keep both alternatives, why not?

BTW did you check what is the case with Arm/RISC-V: Do they have excepting mod/div and do they have vector ISA for integer div?

Reply:
> but if it's easy to keep both alternatives, why not?

downside is maintenance. But ok, i'll restore it.

> did you check what is the case with Arm/RISC-V

ARM has a control register bit to enable/disable trap on div-by-zero. I couldn't find what is the behavior of INT_MIN divided by -1.

RISC-V [does not trap](https://five-embeddev.com/riscv-user-isa-manual/Priv-v1.12/m.html#division-operations) and it seems there is no option to enable trap.